src/api/form/PDFButton.ts

```diff
@@ -181,6 +181,7 @@ export default class PDFButton extends PDFField {
 
     // Add widget to the given page
     page.node.addAnnot(widgetRef);
+    this.updateWidgetPageReference(widget, page.ref);
```

Comment:
I definitely like the idea of setting the page reference for widgets. However, I'd prefer to add an optional `page` property to the `PDFField.createWidget` so it better fits with the existing API. So setting it would then look like:
```js
const widget = this.createWidget({
  width: options?.width ?? 100,
  height: options?.height ?? 50,
  ...,
  hidden: options?.hidden,
  page: page.ref,
});
```

Reply:
@Hopding I have reworked the interface to work as described.

As the function P() was defined on the PDFWidgetAnnotation Class, I added a setP function. This makes it so that PDFField is not directly accessing the widget dictionary, which should reduce coupling.

Now, in each context where createWidget is called, the page reference is passed as an option.

Is there someplace to document this low level API that I should be adding to?